tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,606 @@
+					pvcName,
+				)
+
+				By("deleting the storageClass " + scName)
+				err = ops.SCClient.Delete(context.TODO(), scName, &metav1.DeleteOptions{})
+				Expect(err).To(
+					BeNil(),
+					"while deleting storageclass {%s}",
+					scName,
+				)
+			})
+		})
+
+		When("pod consuming pvc with quota applied created", func() {
+			It("should be up and running and should not be able to write more than the quota limit", func() {
+				By("building a StorageClass")
+				scObj, err = sc.NewStorageClass(
+					sc.WithGenerateName(scNamePrefix),
+					sc.WithLabels(map[string]string{
+						"openebs.io/test-sc": "true",
+					}),
+					sc.WithLocalPV(),
+					sc.WithHostpath(xfsHostpathDir),
+					sc.WithVolumeBindingMode("WaitForFirstConsumer"),
+					sc.WithReclaimPolicy("Delete"),
+					sc.WithParameters(map[string]string{
+						"enableXfsQuota": "true",
+						"softLimitGrace": "0%",
+						"hardLimitGrace": "0%",
+					}),
+				)
+				Expect(err).To(
+					BeNil(),
+					"while building StorageClass with name prefix {%s}",
+					scNamePrefix,
+				)
+
+				By("creating StorageClass API resource")
+				scObj, err = ops.SCClient.Create(context.TODO(), scObj)
+				Expect(err).To(
+					BeNil(),
+					"while creating StorageClass with name prefix {%s}",
+					scNamePrefix,
+				)
+				scName = scObj.ObjectMeta.Name
+
+				By("building a PVC with StorageClass " + scName)
+				pvcObj, err = BuildPersistentVolumeClaim(pvcName, scName, capacity, accessModes)
+				Expect(err).ShouldNot(
+					HaveOccurred(),
+					"while building PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("creating above PVC with StorageClass " + scName)
+				newPvc, err := ops.PVCClient.WithNamespace(namespaceObj.Name).Create(context.TODO(), pvcObj)
+				Expect(err).To(
+					BeNil(),
+					"while creating PVC {%s} in namespace {%s}",
+					pvcName,
+					namespaceObj.Name,
+				)
+
+				By("building a pod with busybox image")
+				// NOTE: this pod has commands to check the correctness of applied quota
+				// which depends on the capacity of the volume used in above PVC.
+				// The command currently can validate quota which is under 3M capacity.
+				// If the PVC capacity is changed, the command needs to be changed accordingly.
+				podObj, err = pod.NewBuilder().
+					WithName(podName).
+					WithNamespace(namespaceObj.Name).
+					WithLabels(labelselector).
+					WithContainerBuilder(
+						container.NewBuilder().
+							WithName("busybox").
+							WithImage("busybox").
+							WithCommandNew(
+								[]string{
+									"/bin/sh",
+									"-c",
+								},
+							).
+							WithArgumentsNew([]string{
+								"sleep 150000;",
+								"for i in $(seq 1 3); do sudo dd if=/dev/zero of=" + "pvc-" + string(newPvc.UID) + "/test.txt bs=1M count=$i 2>/dev/null; if [ $? -ne 0 ]; then echo \"filesize of $i Mb failed, quota reached!\"; exit 1; else echo \"filesize of $i Mb was ok\"; fi; done",
+							}).
```

Comment:
With this, we can't ensure whether capacity is reached or not deploy busy box app and let's [exec](https://github.com/openebs/dynamic-nfs-provisioner/blob/develop/tests/exec.go#L31) into the pod and run the command